### PR TITLE
Simple enunu support

### PR DESCRIPTION
**Pull request by rokujyushi**
## Summary of changes

* Added the ability to load models available in SimpleENUNU.

## Why make this change

* To be able to work with SimpleENUNUServer.
* To be able to load SimpleENUNU compatible models.

## What we did.

* [x] Edited VoicebankLoader to be able to load models available in SimpleENUNU.
* [x] Enabled to read settings not written in config.yaml file.

## Changes.

image

## Scope of impact.

* Loading voice banks.
* EnunuConfig setup

## How to do it.

* Put sound data of SimpleENUNU compatible models in the Voice folder

## Issues

* SingingStyle support

## Remarks

* The following is SimpleENUNUServer.
https://github.com/rokujyushi/SimpleEnunu/releases/tag/SimpleEnunu-OpenUtau-0.1.0
* Based on SimpleENUNU distributed in the following tweet.
https://twitter.com/canon_73/status/1654293958607114240

### CODE REVIEW

File: OpenUtau.Core/Enunu/EnunuConfig.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
 using System.IO;
+using System.Security.Cryptography.X509Certificates;
```
Comment (oxygen-dioxide) [inline OpenUtau.Core/Enunu/EnunuConfig.cs:3]: You may have include System.Security.Cryptography.X509Certificates by mistake. visual studio often includes this when coding even if we don't need it.

File: OpenUtau.Core/Enunu/EnunuConfig.cs [hunk 2]
```diff
@@ -12,10 +13,61 @@ class EnunuConfig {
...
         }
+
+        public static RawEnunuConfig SetSimpleENUNUConfig(string location) {
+            const string tableExtension = ".table";
+            const string hedExtension = ".hed";
+            const string configYaml = "config.yaml";
+            const string modelPath = "model";
+            var config = new RawEnunuConfig();
+
+            if (Directory.Exists(Path.Combine(location, modelPath))) {
```
Comment (stakira) [inline OpenUtau.Core/Enunu/EnunuConfig.cs:33]: Code in `if` and `else` are basically the same. You could instead create an array `[location, location/model]`, then loop through it. Again, the directory existence check is unnecessary when you already check the existence of a file in it.

File: OpenUtau.Core/Classic/VoicebankLoader.cs [hunk 3]
```diff
@@ -92,9 +94,22 @@ public class VoicebankLoader {
                 }
             }
             var enuconfigFile = Path.Combine(dir, kEnuconfigYaml);
+            var simpleEnunuFolder = Path.Combine(dir, kEnunuModelPath);
             if (File.Exists(enuconfigFile)) {
                 voicebank.SingerType = USingerType.Enunu;
-            } else {
+            } 
+            else if (Directory.Exists(simpleEnunuFolder)) {
+                if (File.Exists(Path.Combine(simpleEnunuFolder, kConfigYaml))) {
```
Comment (stakira) [inline OpenUtau.Core/Classic/VoicebankLoader.cs:102]: 1. The directory existence check is unnecessary when you already check file existence. Same for the below one.
2. When you have `if (...) { if (...) { voicebank.SingerType = ... } }`, there is a chance the first condition is true but the second is false. As a result `SingerType` will never be set. This should be avoided.

**Comment by rokujyushi** [comment]
Corrected the part you pointed out. Please check it.

**Comment by rokujyushi** [comment]
Could you check?
## Changes
- Remove unnecessary using
- Fixed VoicebankLoader processing
- Fixed processing of EnunuConfig

**Comment by stakira** [comment]
Thanks!

**Pull request merged**